### DISCUSSION TITLE
Scripting example,  using the Remote as a Proximity Detector (HERO11)

### DISCUSSION THREAD

**Discussion by dnewman-gpsw** [gopro/labs#283] [General]
Youtube demo:  https://www.youtube.com/shorts/iemaVm2cQek

This one could have some serious applications, using the Remote as a wearable proximity camera trigger.  If you have the remote paired with one camera only, it will auto connect with that camera when within range.  Labs now has a testable status for when the remote is connected or not, this can be used to start and stop captures without having to press record on the remote. 

![image](https://user-images.githubusercontent.com/10856084/227628349-7d4bba40-78be-4357-b266-be8ac9388425.png)

Script: oMPMSG="RC trigger"<r0>r:C0"Remote in range"+!S>r0<r:C0"Remote gone"+!E!R19

as pseudo code

```
  change the banner message      // oMPMSG="RC trigger"
loop:
  if(record < 0)                 // <r0    (if not recording)
    if(remoteConnected > 0)      // >r:C0  (and if remote is connected)
      display "remote in range"  // "Remote in range"
      and                        // +
      start                      // !S
  if(record > 0)                 // >r0    (if recording)
    if(remoteConnected < 0)      // <r:C0  (and if remote is not connected)
      display "remote gone"      // "Remote gone"
      and                        // +
      stop capture               // !E
  goto loop                      // !R19    (jump to the 19th character in the script)

```

**Comment by ap0ll0nas** [comment]
Thanks! The remote in range works however when the remote is out of range, the video doesn't stop - tried that 3 times. Furthermore, after about 10min of video, the remote would not come back in range - this is on a location I keep practicing and where this was not an issue before.

**Reply by dnewman-gpsw** [reply]
It is working in my tests.  HERO11 v.2.10.70, not mini?

**Reply by dnewman-gpsw** [reply]
Is the remote updated to v2.0.0?

**Comment by ap0ll0nas** [comment]
Updating the Gopro 11 black firmware did the trick!! Initially it was showing the message leading me to think no update was required.

This is a KILLER feature!! I also love that when you stop the recording with the remote, it immediately starts a new one, allowing you to
fine tune the starting point at the scene if and when you want to, while the in range feature is still on. Well done.

**Comment by andrerav** [comment]
Hm, I have an older remote that I bought with my GoPro 7 a few years ago. Would that work with this? This is definitely a heckin killer feature!

**Reply by dnewman-gpsw** [reply]
The old remote used WiFi, the new Remote is BLE, and this hack uses the BLE connection.  You might have to upgrade https://gopro.com/en/us/shop/mounts-accessories/the-remote/ARMTE-003.html
